VideoApp/VideoApp/Stores/MainParticipant/MainParticipantStore.swift

```diff
@@ -26,59 +45,29 @@ class MainParticipantStore {
         self.room = room
         self.participantsStore = participantsStore
         self.notificationCenter = notificationCenter
-        self.mainParticipant = room.localParticipant
-        update()
-        notificationCenter.addObserver(self, selector: #selector(handleRoomUpdate(_:)), name: .roomUpdate, object: room)
-        notificationCenter.addObserver(self, selector: #selector(handleParticipantUpdate(_:)), name: .participantUpdate, object: nil)
-        notificationCenter.addObserver(self, selector: #selector(handleParticipantsStoreUpdate(_:)), name: .participantsStoreUpdate, object: participantsStore)
-    }
-
-    @objc private func handleRoomUpdate(_ notification: Notification) {
-        guard let payload = notification.payload as? Room.Update else { return }
-        
-        switch payload {
-        case .didStartConnecting, .didConnect, .didFailToConnect, .didDisconnect: break
-        case .didAddRemoteParticipants, .didRemoveRemoteParticipants: update()
-        }
-    }
-    
-    @objc private func handleParticipantUpdate(_ notification: Notification) {
-        guard let payload = notification.payload as? ParticipantUpdate else { return }
-        
-        switch payload {
-        case let .didUpdate(participant):
-            if participant === mainParticipant {
-                postUpdate()
-            }
-            
-            update()
-        }
-    }
-
-    @objc private func handleParticipantsStoreUpdate(_ notification: Notification) {
+        mainParticipant = room.localParticipant
+        videoTrack = mainParticipant.mainVideoTrack
         update()
+        notificationCenter.addObserver(self, selector: #selector(update), name: .roomUpdate, object: room)
+        notificationCenter.addObserver(self, selector: #selector(update), name: .participantUpdate, object: nil)
+        notificationCenter.addObserver(self, selector: #selector(update), name: .participantsStoreUpdate, object: participantsStore)
     }
     
-    private func update() {
+    @objc private func update() {
```

Comment:
Does this still need to be `@objc`?

Reply:
Yeah the notification interface requires it I think because `#selector` relies on some objc magic. 😞